dbms/src/Storages/DeltaMerge/StableValueSpace.cpp

```diff
@@ -461,11 +450,69 @@ SkippableBlockInputStreamPtr StableValueSpace::Snapshot::getInputStream(
     const DMContext & context,
     const ColumnDefines & read_columns,
     const RowKeyRanges & rowkey_ranges,
-    const RSOperatorPtr & filter,
     UInt64 max_data_version,
     size_t expected_block_size,
     bool enable_handle_clean_read,
     ReadTag read_tag,
+    const DMFilePackFilterResults & pack_filter_results,
+    bool is_fast_scan,
+    bool enable_del_clean_read,
+    const std::vector<IdSetPtr> & read_packs,
+    bool need_row_id)
+{
+    LOG_DEBUG(
+        log,
+        "start_ts: {}, enable_handle_clean_read: {}, is_fast_mode: {}, enable_del_clean_read: {}",
+        max_data_version,
+        enable_handle_clean_read,
+        is_fast_scan,
+        enable_del_clean_read);
+    SkippableBlockInputStreams streams;
+    std::vector<size_t> rows;
+    streams.reserve(stable->files.size());
+    rows.reserve(stable->files.size());
+
+    for (size_t i = 0; i < stable->files.size(); ++i)
+    {
+        DMFileBlockInputStreamBuilder builder(context.global_context);
+        builder.enableCleanRead(enable_handle_clean_read, is_fast_scan, enable_del_clean_read, max_data_version)
+            .enableColumnCacheLongTerm(context.pk_col_id)
+            .setDMFilePackFilterResult(!pack_filter_results.empty() ? pack_filter_results[i] : nullptr)
+            .setColumnCache(column_caches[i])
+            .setTracingID(context.tracing_id)
+            .setRowsThreshold(expected_block_size)
+            .setReadPacks(read_packs.size() > i ? read_packs[i] : nullptr)
+            .setReadTag(read_tag);
+
+        streams.push_back(builder.build(stable->files[i], read_columns, rowkey_ranges, context.scan_context));
+        rows.push_back(stable->files[i]->getRows());
+    }
+    if (need_row_id)
+    {
+        return std::make_shared<ConcatSkippableBlockInputStream</*need_row_id*/ true>>(
+            streams,
+            std::move(rows),
+            context.scan_context);
+    }
+    else
+    {
+        return std::make_shared<ConcatSkippableBlockInputStream</*need_row_id*/ false>>(
+            streams,
+            std::move(rows),
+            context.scan_context);
+    }
+}
+
+SkippableBlockInputStreamPtr StableValueSpace::Snapshot::tryGetInputStreamWithVectorIndex(
```

Comment:
Why do we need to add this method? Seems it is the same as `StableValueSpace::Snapshot::getInputStream` except ann_query_info is not nullptr?

Reply:
I will continue to separate all `tryGetInputStreamWithVectorIndex` in the next PR, to make it more clear.